cli/docker-machine.go

```diff
@@ -0,0 +1,26 @@
+package run
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func getDockerMachineIp() (string, error) {
+	machine := os.ExpandEnv("$DOCKER_MACHINE_NAME")
```

Comment:
I think calling `docker-machine active` to get the machine name would be more reliable as environmental variables might not be set or overridden by users.

Also we're calling `docker-machine ip` instead of DOCKER_HOST already anyways.

Reply:
Just tried doing `docker-machine active` and didn't get any output (even though one of the machines is running). I am not familiar with this command can you describe more of what needs to be done ?

Reply:
Oh interesting.  Mine outputs `default` and the explanation is that

```
$ docker-machine
...
Commands:
  active                Print which machine is active
..
```

What's your output of `docker-machine env`? Does it have this line? `SET DOCKER_MACHINE_NAME=default`

Reply:
Output for `docker-machine env dvm` "dvm" is the name of the docker-machine

```
PS C:\WINDOWS\system32> docker-machine env dvm
$Env:DOCKER_TLS_VERIFY = "1"
$Env:DOCKER_HOST = "tcp://172.17.209.142:2376"
$Env:DOCKER_CERT_PATH = "C:\Users\user\.docker\machine\machines\dvm"
$Env:DOCKER_MACHINE_NAME = "dvm"
$Env:COMPOSE_CONVERT_WINDOWS_PATHS = "true"
# Run this command to configure your shell:
# & "C:\Program Files\Docker\Docker\Resources\bin\docker-machine.exe" env dvm | Invoke-Expression
```

Reply:
How about `docker-machine ls` Does "dvm" line has an active mark on the left?

Reply:
Yes

Reply:
That's really strange then.  `docker-machine ls` reports `dvm` is active while `docker-machine active` reports nothing is active.  I can't think of anything else than it's a docker-machine bug.

Reply:
I was able to find similar issues in their repo.  https://github.com/docker/machine/issues/1651 https://github.com/docker/machine/issues/3448

Reply:
Good catch, definitely that issue. What do you suggest ?

Reply:
Well, I guess we don't have any choice other than DOCKER_HOST.  I just looked up their active check, and it's also just a combination of [DOCKER_HOST + current status] https://github.com/docker/machine/blob/19973f2b1bbcb7b1c07a3de572177bcaf8dedcd5/commands/ls.go#L490

Also, docker-machine repository hasn't been updated for several months so we can't expect it to be fixed anytime soon.

Reply:
Think we should leave it like this for now or create a new issue regarding this.